core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -83,6 +83,32 @@ object Animation {
   /** Create an animation builder from a sequence of starting values */
   def startWithSequence[A: ToRefTree](start: Seq[A]) = Builder(start.toVector)
 
+  private implicit class TakeUntilStream[T](s: Stream[T]) {
+    def takeWhileInclusive(p: T => Boolean): Stream[T] =
+      if (!s.isEmpty && p(s.head)) Stream.cons(s.head, s.tail takeWhileInclusive p)
+      else Stream.cons(s.tail.head, Stream.empty)
+  }
+
+  /** Create an animation builder from a seed until a fixpoint (f(x) == x) is reached */
+  def iterateUntilFixPoint[A: ToRefTree](seed: A)(iteration: (A => A))(
+    max: Int = 10) =
+    Builder(
+      Stream
+        .iterate(seed)(iteration)
+        .take(max)
+        .takeWhileInclusive(x => x != iteration(x))
+        .toVector)
+
+  /** Create an animation builder from a seed until a a given predicate yields true */
+  def iterateUntil[A: ToRefTree](seed: A)(iteration: (A => A))(
```

Comment:
Yeah, on a second thought I agree. What about `while`/`until`? How does that play with your use-cases? I am considering if the predicate should come first, since it is logical for it to follow the `while` word, and the iteration function is probably more complex, so it should be more `{}`-friendly.